BDArmory/Control/BDModuleSurfaceAI.cs

```diff
@@ -390,27 +441,179 @@ void PilotLogic()
                                     continue;
                             }
                             dodgeVector = PredictCollisionWithVessel(vs.Current, 5f * predictMult, 0.5f);
-                            if (dodgeVector != null) break;
+                            if (dodgeVector != Vector3.zero) // Dodge the first potential collision (this isn't necessarily the closest, but multi-vessel collisions are unlikely).
+
+
+                            {
+                                vesselCollision = true;
+                                collidingWith = vs.Current.GetName();
+                                break;
+                            }
+                        }
+                    }
+
+                    { // Terrain/building collisions  FIXME We're only checking buildings, should we drop that and check terrain too?
```

Comment:
Any thoughts on this?

Reply:
Wouldn't that be handled by the terrain pathfind waypoint grid stuff elsewhere in the code?
I have very little experience with the srf AI's terrain navigation (since most of my tests have been flat areas like the ice floes or KSC, or near flat with gentle rolling hills), I'd need to run some test vehicles around the hillier areas of Kerbin and see how they do first.
I think it would require special handling (at minimum something like ``if (Vector3.Angle(hit.normal, GetUpDirection(hit.position) < maxSlope)``); at current the AI would want to veer away the moment it sees raised terrain ahead of it, but is that a hill or mountain that needs to be steered away from, or a gentle incline the vehicle can drive up/over?

Reply:
> Wouldn't that be handled by the terrain pathfind waypoint grid stuff elsewhere in the code?

No idea. I haven't really looked at the pathfind stuff, so I'm not really sure how that's done. It's probably sufficient to just have it doing it for buildings for now anyway. I don't think there's any real cliff faces for vehicles to run into that cause problems like buildings do.